src/lib/viewers/media/DashViewer.js

```diff
@@ -488,17 +486,68 @@ class DashViewer extends VideoBaseViewer {
             textCues.push(new shaka.text.Cue(start, end, text));
         });
 
+        return textCues;
+    }
+
+    /**
+     * Loads auto-generated captions
+     *
+     * @public
+     * @param {Object} [skillsData] - optional skills data used to generate captions
+     * @return {void}
+     */
+    loadAutoGeneratedCaptions(skillsData) {
+        const skillsCards = skillsData || getProp(this.options, 'file.metadata.global.boxSkillsCards.cards');
+        // Do nothing if we don't have skills data or already have captions
+        if (!skillsCards || this.autoCaptionDisplayer) {
+            return;
+        }
+
+        // Save passed in skills data to avoid regenerating captions unnecessarily
+        this.skillsData = skillsData;
+
+        const textCues = this.parseTextCues(skillsCards);
+
         if (textCues.length > 0) {
             this.autoCaptionDisplayer = new shaka.text.SimpleTextDisplayer(this.mediaEl);
             this.autoCaptionDisplayer.append(textCues);
             this.player.configure({ textDisplayFactory: this.autoCaptionDisplayer });
+            // Update the subtitles/caption button to reflect auto-translation
+            this.mediaControls.setLabel(this.mediaControls.subtitlesButtonEl, __('media_auto_generated_captions'));
             this.mediaControls.initSubtitles(
                 [__('auto_generated')],
                 getLanguageName(this.options.location.locale.substring(0, 2))
             );
+        }
+    }
 
-            // Update the subtitles/caption button to reflect auto-translation
-            this.mediaControls.setLabel(this.mediaControls.subtitlesButtonEl, __('media_auto_generated_captions'));
+    /**
+     * Edits auto-generated captions
+     *
+     * @public
+     * @param {Object} skillsData - updated skills data
+     * @return {void}
+     */
+    editAutoGeneratedCaptions(skillsData) {
+        // Do nothing if we don't have skills data
+        if (!skillsData || skillsData === this.skillsData) {
+            return;
+        }
+
+        // Save passed in skills data to avoid regenerating captions unnecessarily
+        this.skillsData = skillsData;
+
+        const textCues = this.parseTextCues(skillsData);
+
+        if (textCues.length > 0) {
+            const areAutoCaptionsVisible = this.autoCaptionDisplayer.isTextVisible();
+            this.autoCaptionDisplayer.destroy();
+            this.autoCaptionDisplayer = new shaka.text.SimpleTextDisplayer(this.mediaEl);
+            this.autoCaptionDisplayer.append(textCues);
+            this.player.configure({ textDisplayFactory: this.autoCaptionDisplayer });
+            if (areAutoCaptionsVisible) {
+                this.autoCaptionDisplayer.setTextVisibility(true);
```

Comment:
Could we call `this.autoCaptionDisplayer.setTextVisibility(areAutoCaptionsVisible)` to clarify that we're preserving the user's preference? What do you think?

Reply:
I don't think there's any harm to calling it again with false if they are already off, and it'll make the code cleaner.